sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorSession.java

```diff
@@ -525,14 +528,18 @@ protected ReactorReceiver createConsumer(String entityPath, Receiver receiver,
 
         final Sender sender = session.sender(linkName);
         sender.setTarget(target);
-
-        final Source source = new Source();
-        sender.setSource(source);
         sender.setSenderSettleMode(SenderSettleMode.UNSETTLED);
 
+        final Source source = new Source();
         if (linkProperties != null && linkProperties.size() > 0) {
+            String clientIdentifier = (String) linkProperties.get(CLIENT_IDENTIFIER);
+            if (CoreUtils.isNullOrEmpty(clientIdentifier)) {
```

Comment:
Is this condition wrong?

Reply:
Yes, the condition is wrong. About the test, I've already ask Jesse and Anu, we'll run a long-running tests for these changes , because once we get into the AMQP level of things, we lose testability on many things like this.